### PR TITLE
Check return value of connect to trap sooner

### PR DESCRIPTION
**Pull request by c1728p9**
Check the return value of connect and error if it did not succeed. This make socket errors more obvious since the TLS handshake has not started at this point. This is to help more quickly identify socket issues like in #17.

Error message before
```
Using Ethernet LWIP
Client IP Address is 192.168.1.69
Connecting with developer.mbed.org
Starting the TLS handshake...
mbedtls_ssl_handshake() failed: -0x0001 (-1): UNKNOWN ERROR CODE (0001)
MBED: Socket Error: -1
```

Error message after
```
Using Ethernet LWIP
Client IP Address is 192.168.1.69
Connecting with developer.mbed.org
Failed to connect
MBED: Socket Error: -1
```

### CODE REVIEW

**Comment by c1728p9** [comment]
Can you provide more details on how to fix the style? I tried to match the rest of the file in this PR.

File: tls-client/main.cpp [hunk 1]
```diff
@@ -207,7 +207,12 @@ class HelloHTTPS {
 
         /* Connect to the server */
         mbedtls_printf("Connecting with %s\r\n", _domain);
-        _tcpsocket->connect( _domain, _port );
+        ret = _tcpsocket->connect( _domain, _port );
+        if (ret != NSAPI_ERROR_OK) {
+            mbedtls_printf("Failed to connect\r\n");
+            onError(_tcpsocket, -1 );
```
Comment (andresag01) [inline tls-client/main.cpp]: Style: remove additional space after `-1`

File: tls-client/main.cpp [hunk 2]
```diff
@@ -207,7 +207,12 @@ class HelloHTTPS {
 
         /* Connect to the server */
         mbedtls_printf("Connecting with %s\r\n", _domain);
-        _tcpsocket->connect( _domain, _port );
+        ret = _tcpsocket->connect( _domain, _port );
```
Comment (andresag01) [inline tls-client/main.cpp]: Style: Please remove space after `(` and before `)`.

**Comment by c1728p9** [comment]
Hi @andresag01 I updated the PR to remove the additional spaces.

**Comment by sbutcher-arm** [comment]
Reviewed, approved and tested by @andresag01, so can be merged.

**Pull request merged**